src/reporting/clients/ODataClient.ts

```diff
@@ -0,0 +1,144 @@
+    const segments = odataItem.url.split("/");  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        "odataItem",
+        "odata item was invalid.",
+      );
+    }
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/${odataItem.url}`;
+    const request = this.createRequest("GET", accessToken);
+    return new EntityListIteratorImpl(async () => getEntityCollectionPage<ODataEntityValue>(
+      url,
+      async (nextUrl: string): Promise<Collection<ODataEntityValue>> => {
+        const response: ODataEntityResponse = await this.fetchJSON<ODataEntityResponse>(nextUrl, request);
+        const link: PagedResponseLinks = {
+          self: {
+            href: nextUrl,
+          },
+        };
+        if(response["@odata.nextLink"]) {
+          link.next = {
+            href: response["@odata.nextLink"],
+          };
+        }
+        return {
+          values: response.value,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          _links: link,
+        };
+      }));
+  }
+
+  public async getODataReportMetadata(accessToken: AccessToken, reportId: string): Promise<ODataTable[]> {
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/$metadata`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    const response = await this.fetchData(url, requestOptions);
+    return this.parseXML(response);
+  }
+
+  private async parseXML(response: Response): Promise<ODataTable[]> {
+    const options = {
+      ignoreAttributes: false,
+      attributeNamePrefix : "",
+      transformTagName: (tagName: string) => tagName.charAt(0).toLowerCase() + tagName.slice(1),
+    };
+    const parser = new XMLParser(options);
+    const text = await response.text();
+    const parsedXML = parser.parse(text);
+    if (!parsedXML["edmx:Edmx"]["edmx:DataServices"].hasOwnProperty("schema") || !(parsedXML["edmx:Edmx"]["edmx:DataServices"].schema instanceof Array)) {
+      return [];
+    }
+
+    const schemas = parsedXML["edmx:Edmx"]["edmx:DataServices"].schema;
+    let entitySet: Array<ODataMetaDataEntitySet>;
+    for (const schema of schemas) {
+      if (schema.Namespace === "Default") {
+        entitySet = this.makeArray(schema.entityContainer.entitySet);
+        break;
+      }
+    }
+
+    const result: ODataTable[] = [];
+    for (const entity of entitySet!) {
+      result.push(this.getEntitySetByName(entity, schemas));
+    }
+    return result;
+  }
+
+  private getEntitySetByName(entity: ODataMetaDataEntitySet, schemas: ODataMetaDataSchema[]): ODataTable {
+    const table: ODataTable = {
+      name: entity.Name,
+      columns: [],
+    };
+    const identifiers = entity.EntityType.split(".");
+
+    const entityTypes: Array<ODataMetaDataEntityType> = [];
+    schemas.filter((schema: ODataMetaDataSchema) => {return schema.Namespace === identifiers[0];})
```

Comment:
`schemas.filter((s) => s.Namespace === identifiers[0])` - takes up less space. Also the formatting was off on this line.

Reply:
changed